bool left
bool front
bool right
bool far_front